From cd94a5c750554f21ddbff37f53ff629128200259 Mon Sep 17 00:00:00 2001
From: "Michael S. Tsirkin" <mst@redhat.com>
Date: Wed, 6 Nov 2024 17:29:35 -0500
Subject: [PATCH 2/3] virtio-net: disable USO for RHEL9

RH-Author: MST <mst@redhat.com>
RH-MergeRequest: 289: Disable USO for virtio-net to fix RHEL10 to RHEL9 migration
RH-Jira: RHEL-40950
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Commit: [1/1] 9fc588b66c5bad72df1e9b644f48a2a7aa82bfa3 (mstredhat/qemu-kvm-centos)

Theoretically, QEMU from RHEL9.3 and on supports USO,
but practically we clear the support because RHEL9
kernels do not support that.

Now that RHEL10 beta does we suddenly get a migration compatibility
issue. We should not have enabled the feature in RHEL9 userspace,
but luckily, it's not too late to fix that.

Note: if we ever change RHEL9 kernel to enable USO, we will need to
mask this in RHEL9 QEMU, too.

Upstream status: n/a: upstream has no guarantee if kernel features change
Tested: lightly on developer's machine.
JIRA: https://issues.redhat.com/browse/RHEL-40950
Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
---
 hw/core/machine.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hw/core/machine.c b/hw/core/machine.c
index d95f246f66..04d180eac4 100644
--- a/hw/core/machine.c
+++ b/hw/core/machine.c
@@ -337,6 +337,11 @@ GlobalProperty hw_compat_rhel_9_5[] = {
     { TYPE_VIRTIO_IOMMU_PCI, "aw-bits", "64" },
   /* hw_compat_rhel_9_5 from hw_compat_8_2 */
     { "virtio-gpu-device", "x-scanout-vmstate-version", "1" },
+    /* supported by userspace, but RHEL 9 *kernels* do not support USO. */
+    /* TODO: if we ever add 9.6 compat, this has to be there, too */
+    { TYPE_VIRTIO_NET, "host_uso", "off"},
+    { TYPE_VIRTIO_NET, "guest_uso4", "off"},
+    { TYPE_VIRTIO_NET, "guest_uso6", "off"},
 };
 const size_t hw_compat_rhel_9_5_len = G_N_ELEMENTS(hw_compat_rhel_9_5);
 
-- 
2.39.3

